### PR TITLE
[front] enh: add browse action

### PR DESCRIPTION
**Pull request by tdraier**
https://github.com/dust-tt/dust/issues/5469
https://github.com/dust-tt/dust/issues/5470
https://github.com/dust-tt/dust/issues/5471
https://github.com/dust-tt/dust/issues/5473

## Description

Adds a new browse action to fetch a list of urls.

## Risk

- New action is hidden in feature flag, 
- Code can be rolled back

## Deploy Plan

Deploy new table in migration, then deploy `front`

### CODE REVIEW

File: types/src/front/assistant/actions/browse.ts
Comment (philipperolet) [inline types/src/front/assistant/actions/browse.ts:15]: what about including the browsed URLs in the schema? so that it can be known which browsed data is which url definitely (without having to rely on mapping to the urls string[]) -- for us in code (e.g. if we move the array around) and for the model when rendering the action result

File: front/lib/models/assistant/actions/browse.ts
Comment (philipperolet) [inline front/lib/models/assistant/actions/browse.ts:93]: here we should do `string[]` IMHO, just like e.g. tags from [process action model](https://github.com/dust-tt/dust/blob/932b70f9409fb904cc5537f7d5ef5bd8bed82334/front/lib/models/assistant/actions/process.ts#L21)

**Review by philipperolet — COMMENTED** [review]
Looks great :+1: Nice work

Left 2 comments

**Comment by philipperolet** [comment]
As discussed IRL, testing before merge requires the credentials part in prod (+ maybe some other stuff) => a separate PR for that, then test, then we can merge this one pending resolution of above coms

File: types/src/front/assistant/actions/browse.ts
Comment (tdraier) [reply types/src/front/assistant/actions/browse.ts:15]: Actually already returned by the dust app, just forgot to add it to the schema :-)

**Comment by tdraier** [comment]
> As discussed IRL, testing before merge requires the credentials part in prod (+ maybe some other stuff) => a separate PR for that, then test, then we can merge this one pending resolution of above coms

Here : https://github.com/dust-tt/dust/pull/5597/commits/e771b9456ab1865a7a4ecc76d11976d0649f426f 

only credentials required, also added migration to create table before merging this one

File: front/lib/models/assistant/actions/browse.ts
Comment (tdraier) [reply front/lib/models/assistant/actions/browse.ts]: Nice, I did not know about postgres arrays

**Review by philipperolet — APPROVED** [review]
:+1: :rocket: 

(assuming you could test it well locally)

File: types/src/front/assistant/actions/browse.ts
Comment (philipperolet) [inline types/src/front/assistant/actions/browse.ts:16]: as discussed [in thread](https://dust4ai.slack.com/archives/C050SM8NSPK/p1718191538274749) IMHO here we should have an optional error field (and null content if so)
ip & port are not necessary I think 
url should be the exact url requested as input (maybe it should be a map instead of an array)
if there's a redirect, this should be added somewhere in the return

**Review by philipperolet — COMMENTED** [review]
additional com after the fact, can be handled in a separate PR

File: front/components/assistant_builder/actions/BrowseAction.tsx
Comment (spolu) [inline front/components/assistant_builder/actions/BrowseAction.tsx:12]: Get the content of a *web* page

File: front/lib/api/assistant/actions/utils.ts
Comment (spolu) [inline front/lib/api/assistant/actions/utils.ts:68]: Get the content of a *web* page

File: types/src/front/assistant/actions/browse.ts
Comment (spolu) [inline types/src/front/assistant/actions/browse.ts:17]: status_code to be more explicit?

File: front/lib/api/assistant/actions/browse.ts
Comment (spolu) [inline front/lib/api/assistant/actions/browse.ts:111]: Get *the* content of a *web* page

File: front/lib/api/assistant/actions/browse.ts
Comment (spolu) [inline front/lib/api/assistant/actions/browse.ts:115]: We should type this as a JSON array instead of a string that includes JSON which will be highly misleading and error prone for models to properly escape and format

**Review by spolu — COMMENTED** [review]
Looks great. Left a few comments to be actioned :pray:

**Pull request merged**
